devtools/maven/src/main/java/io/quarkus/maven/BuildMojo.java

```diff
@@ -187,11 +187,13 @@ public void execute() throws MojoExecutionException {
                             .setConfigDir(outputDirectory.toPath())
                             .setBuildSystemProperties(realProperties).build());
             Artifact original = project.getArtifact();
-            if (result.getJar().isUberJar() && result.getJar().getOriginalArtifact() != null) {
-                original.setFile(result.getJar().getOriginalArtifact().toFile());
-            }
-            if (result.getJar().isUberJar()) {
-                projectHelper.attachArtifact(project, result.getJar().getPath().toFile(), "runner");
+            if (result.getJar() != null) {
```

Comment:
I'm not sure if the issue is the null test or the fact that `result.getJar()` is empty. When using XML config, it's not as we don't have the NPE so I think we need Stuart to say if it's normal or not.

Reply:
This is ok, the NativeImageMojo does not attempt to look at the jar result.